docs/aws/audit/esmonitoring/rules/es_ri_payment_pending_remediation.mdx

```diff
@@ -1,6 +1,252 @@
+    check_reserved_instance_status()
+```
+
+By following these steps, you can proactively monitor and prevent Elasticsearch Reserved Instances from remaining in a "payment-pending" status, ensuring that your instances are properly configured and operational.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Log in to the AWS Management Console and navigate to the Elasticsearch service.
+2. In the Elasticsearch dashboard, click on "Reserved Instances" in the left-hand navigation pane.
+3. In the Reserved Instances page, you will see a list of all your reserved instances. Look for the "Status" column in the table.
+4. Check the status of each reserved instance. If any of them have the status "Payment Pending", it indicates that the payment for that reserved instance is not yet complete.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start, you need to install the AWS CLI on your local machine. You can do this by downloading the appropriate installer from the AWS CLI website. Once installed, you can configure it by running `aws configure` and providing your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. List all Elasticsearch Reserved Instances: Use the `es:describe-reserved-elasticsearch-instances` command to list all the Elasticsearch Reserved Instances. The command is as follows:
+
+   ```
+   aws es describe-reserved-elasticsearch-instances
+   ```
+
+3. Filter instances with Payment Pending status: The output of the above command will be a JSON object containing details about all the Elasticsearch Reserved Instances. You need to parse this JSON object to filter out instances that have the status 'payment-pending'. You can use the `jq` command-line JSON processor for this. The command is as follows:
+
+   ```
+   aws es describe-reserved-elasticsearch-instances | jq '.ReservedElasticsearchInstances[] | select(.State=="payment-pending")'
+   ```
+
+4. Check the output: If the above command returns any instances, it means that there are Elasticsearch Reserved Instances with Payment Pending status. If it doesn't return anything, it means that there are no such instances.
+</Accordion>
+
+<Accordion title='Using Python'>
+To detect if an Elasticsearch Reserved Instance has a status of "Payment Pending" using Python scripts, you can use the Boto3 library, which allows you to directly interact with AWS services, including Elasticsearch. Here are the steps:
+
+1. **Install and Configure Boto3:**
+   First, you need to install and configure Boto3 if you haven't done so already. You can install it using pip:
+   ```
+   pip install boto3
+   ```
+   Then, configure your AWS credentials. You can do this by setting the following environment variables:
+   ```
+   AWS_ACCESS_KEY_ID = 'your_access_key'
+   AWS_SECRET_ACCESS_KEY = 'your_secret_key'
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>